cfme/physical/physical_server.py

```diff
@@ -154,6 +160,66 @@ def wait_for_delete(self):
             fail_func=view.browser.refresh
         )
 
+    def validate_stats(self, ui=False):
+        """ Validates that the detail page matches the physical server's information.
+
+        This method logs into the provider using the mgmt_system interface and collects
+        a set of statistics to be matched against the UI. An exception will be raised
+        if the stats retrieved from the UI do not match those retrieved from wrapanapi.
+        """
+
+        # Make sure we are on the physical server detail page
+        if ui:
+            self.load_details()
+
+        # Retrieve the data from the provider
+        providers_data = conf.cfme_data.get("management_systems", {})
+        providers = providers_data
+
+        # Gather the data necessary to instantiate an instance of the management class
+        # defined in wrapanapi.
+        provider_data = providers['lenovo']
+        credentials = provider_data['credentials']
+        credentials = conf.credentials[credentials]
+        provider_kwargs = provider_data.copy()
+        provider_kwargs.update(credentials)
+        provider_kwargs['logger'] = logger
+
+        # Create an instance of the management class
+        mgmt = self.mgmt_class(**provider_kwargs)
+
+        # Check that the stats match
+        self._check_for_matching_stats(mgmt, self.STATS_TO_MATCH, ui=ui)
+
+    def _check_for_matching_stats(self, client, stats_to_match=None, ui=False):
+        """ A function that checks that the stats from CFME and wrapanapi match.
```

Comment:
If you move `data` and `mgmt` properties out of `validate_stats`, I think you can collapse this `_method` and just make it validate_stats - since that method isn't doing anything other than getting data/mgmt and passing mgmt to this.